tests/backup/backup_restore_basic_test.go

```diff
@@ -2763,3 +2764,458 @@ var _ = Describe("{BackupRestoreOnDifferentK8sVersions}", func() {
+				log.InfoD("switched context to destination")
+
+				destinationClusterConfigPath, err := GetDestinationClusterConfigPath()
+				log.FailOnError(err, "failed to get kubeconfig path for destination cluster. Error: [%v]", err)
+
+				err = Inst().S.SetConfig(destinationClusterConfigPath)
+				log.FailOnError(err, "failed to switch to context to destination cluster [%v]", destinationClusterConfigPath)
+
+				ver, err := k8s.ClusterVersion()
+				log.FailOnError(err, "failed to get destination cluster version")
+				destVer, err = semver.Make(ver)
+				log.FailOnError(err, "failed to get destination cluster version")
+			})
+
+			Step("Compare Source and Destination cluster version numbers", func() {
+				log.InfoD("source cluster version: %s ; destination cluster version: %s", srcVer.String(), destVer.String())
+				isValid := srcVer.LT(destVer)
+				dash.VerifyFatal(isValid, true,
+					"source cluster kubernetes version should be lesser than the destination cluster kubernetes version.")
+			})
+
+			log.InfoD("switching to default context")
+			err := SetClusterContext("")
+			log.FailOnError(err, "failed to SetClusterContext to default cluster")
+		})
+
+		Step("deploy the applications on Src cluster", func() {
+			log.InfoD("deploy the applications on Src cluster")
+
+			Step("deploy applications", func() {
+				log.InfoD("deploy applications")
+
+				log.InfoD("switching to source context")
+				err := SetSourceKubeConfig()
+				log.FailOnError(err, "failed to switch to context to source cluster")
+
+				log.InfoD("scheduling applications")
+				scheduledAppContexts = make([]*scheduler.Context, 0)
+				for i := 0; i < Inst().GlobalScaleFactor; i++ {
+					taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+					appContexts := ScheduleApplications(taskName)
+					for _, appCtx := range appContexts {
+						appCtx.ReadinessTimeout = appReadinessTimeout
+						namespace := GetAppNamespace(appCtx, taskName)
+						appCtx.ScheduleOptions.Namespace = namespace
+						scheduledAppContexts = append(scheduledAppContexts, appCtx)
+					}
+				}
+			})
+
+			Step("Validate applications", func() {
+				ValidateApplications(scheduledAppContexts)
+
+				log.InfoD("switching to default context")
+				err := SetClusterContext("")
+				log.FailOnError(err, "failed to SetClusterContext to default cluster")
+			})
+
+			log.InfoD("waiting (for 2 minutes) for any CRs to finish starting up.")
+			time.Sleep(time.Minute * 2)
+			log.Warnf("no verification is done; it might lead to undetectable errors.")
+		})
+
+		Step("Creating rules for backup", func() {
+			log.InfoD("creating pre rule for deployed apps")
+			for i := 0; i < len(appList); i++ {
+				preRuleStatus, ruleName, err := Inst().Backup.CreateRuleForBackup(appList[i], orgID, "pre")
+				log.FailOnError(err, "creating pre rule for deployed apps failed")
+				dash.VerifyFatal(preRuleStatus, true, "verifying pre rule for backup")
+
+				if ruleName != "" {
```

Comment:
I think if there is no pre or post exec rule Inst().Backup.CreateRuleForBackup will return "" for ruleName.

Reply:
I'm not sure what you're asking me to do here. I just took this code from BasicBackupCreation

Reply:
Remove this whole step. It is not needed.

Reply:
✅